### DISCUSSION TITLE
Unity webGL project not working with git lfs!

### DISCUSSION THREAD

**Discussion by Jinzo420** [git-lfs/git-lfs#5047] [Q&A]
Hi all, 

I have a unity game which when under 100mb and uploaded normally to github runs perfect, but after it gets over 100mb and i use git lfs for upload my webgl project just freezes after the unity loading icon. Giving me these console errors. 

no boot config - using default values

_JS_Log_Dump @ metaTKnew.framework.js:2
[libil2cpp] ERROR: Could not open Il2CppData/Metadata/global-metadata.dat
RuntimeInitializeOnLoadManagerInitializer: Failed reading 'RuntimeInitializeOnLoads.json'
Failed reading 'ScriptingAssemblies.json'
No GlobalGameManagers file was found at , quitting player!
Failed to initialize player

I've tried numerous things with no luck.. Could anyone point me in right direction? Would be much appreciated. 

Thanks

**Comment by bk2204** [comment]
Hey,

My guess is that for whatever reason, `git lfs install` hasn't been run, and as a result, what you have in your tree are the pointer files instead of the large files themselves.  If you haven't run `git lfs install`, then do that, and then you should be able to get the large files in your working tree by running `git lfs pull`.

**Reply by Jinzo420** [reply]
Thanks for the reply. It appears the large file is in my project (not the pointer files) and unfortunately the same error occurs. I still feel i'm missing something obvious, maybe in the method I upload the large file? here's a few screenshots of the big file itself in my project. Does it look right to you? I appreciate your reply. Thanks


![Screen Shot 2022-06-16 at 12 52 11 PM](https://user-images.githubusercontent.com/83331345/173967432-1ab7599f-d489-4232-90a0-344890adf290.png)
![Screen Shot 2022-06-16 at 12 52 25 PM](https://user-images.githubusercontent.com/83331345/173967484-b9a4b5d3-801e-46e6-ba04-4657ae5b3611.png)
![Screen Shot 2022-06-16 at 12 51 45 PM](https://user-images.githubusercontent.com/83331345/173967513-1ad7e39f-5d44-4f71-a7f0-5a616f5bbbcc.png)
.

**Reply by bk2204** [reply]
Hey,

This shows that the object is stored using Git LFS, which is correct.  However, that means that in the repository, it's stored as a pointer, which is then converted to the large file on checkout.  If in your case your JSON files (look in a text editor) look like this:

```
version https://git-lfs.github.com/spec/v1
oid sha256:917df3320d778ddbaa5c5c7742bc4046bf803c36ed2b050f30844ed206783469
size 3
```

instead of a JSON file, then what you have in the tree is the pointer file.  The particular situation is relevant to your working tree, and not to what GitHub has in the web interface.

If you have a pointer file, then you need to run `git lfs pull` (or, if you haven't pushed the data), `git lfs checkout`, and that should fix the problem.  If it doesn't, please run the command again as `GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1 git lfs pull` (in a POSIX shell, or, on Windows, Git Bash) and paste the output here as an attachment.

If what you have in your working tree is the large file, then this is probably not a Git LFS issue and probably lies elsewhere.

One final note: it looks like you're storing build artifacts in your repository, which is almost certainly a bad idea.  Build artifacts tend to be large and trying to merge them can cause merge conflicts which are hard to resolve.

**Reply by Fraccas** [reply]
Hi @bk2204, 

I'm having the same issue. 

When testing locally everything works as expected, however, once deployed to Git/Heroku I have the same result as the OP. 

I'm using the following Heroku buildpack:
https://elements.heroku.com/buildpacks/radian-software/heroku-buildpack-git-lfs

Deployed App
![image](https://user-images.githubusercontent.com/3107701/175793551-958e419d-d902-473e-8105-83828fcbf645.png)

Locally, the file is in its normal state and not a pointer, however, when deployed it finds the file in question, but fails to read it. 

![image](https://user-images.githubusercontent.com/3107701/175793809-49e6ce5e-052a-4053-99f4-1fb2a1cb4ccb.png)


I'm not sure how to get my deployed app to read the actual binary file rather than the pointer. 

EDIT:
I was actually able to find the hosted file directly through GitHub and point the site directly to it, though this method seems to be extremely slow so we'll likely be looking for an alternative solution.
![image](https://user-images.githubusercontent.com/3107701/175834451-2a141bf1-f133-4c78-b7a7-ca5455b95c6f.png)

**Reply by bk2204** [reply]
Hey,

Wherever you're deploying this, you need to make sure that Git LFS is installed and that you've run `git lfs install` so that it works properly.  Since Heroku doesn't seem to support Git LFS automatically, it looks like you may need to use [this buildpack](https://github.com/radian-software/heroku-buildpack-git-lfs) or something similar.

**Reply by Fraccas** [reply]
Yeah, that is the buildpack that I was using so I attempted to get in touch with them without any luck. 

As Heroku does not have native support for LFS we've switched to hosting our large files on AWS.
